components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -415,7 +415,12 @@ public static String getErrorPageURL(HttpServletRequest request, String errorCod
         if (request == null) {
             return redirectURL;
         }
-        return getRedirectURL(redirectURL, request);
+        if (isAllowAdditionalParamsFromErrorUrlEnabled()) {
+            // Appending additional parameters if the <AllowAdditionalParamsFromErrorUrl> config is enabled
+            return getRedirectURL(redirectURL, request);
+        } else {
+            return redirectURL;
+        }
```

Comment:
suggestion for line 414 to 423:

```
        String redirectURL = getErrorPageURL(errorCode, errorMessage, appName);
        if (request == null || !isAllowAdditionalParamsFromErrorUrlEnabled()) {
            return redirectURL;
        }
        // Appending additional parameters if the <AllowAdditionalParamsFromErrorUrl> config is enabled
        return getRedirectURL(redirectURL, request);
```

Reply:
Suggested change added with e0a7995beb19d81a39eff11df56bc838eac1b687